### PR TITLE
As a developer I want to have our db scructure in project

### PR DESCRIPTION
**Pull request by OlegKarapysh**
Added all entities from db schema and relationships between them.
Currently all entities have DeleteBehavior.NoAction, it can be changed if needed.

### CODE REVIEW

**Review by Anatolii-Ivanov — DISMISSED** [review]
For me it looks ok, not sure if we need so many lists, but as they are empty, it's ok

**Comment by MaxShelestiuk** [comment]
1) I think we should change table naming to singular or plural for all tables (for now they are different):
image

2) @ABertashDev and me found a mistake in DB Schema: 
2.1 Change FK to Commits
2.2 Change field CommitParentId to CommitId
image

3) JoinEntities doesn`t have Columns for PrimaryKey, composite keys are used instead. It is not bad, but If you have composite primary keys, anyone joining to your table from a child table must "drag along" all those columns, and all those columns are also going to be present in the child table and the JOIN statements are pretty messy. Much better to have a single key for the JOIN
image

**Comment by MaxShelestiuk** [comment]
1) I think we should change table naming to singular or plural for all tables (for now they are different):
image

2) @ABertashDev and me found a mistake in DB Schema: 
2.1 Change FK to Commits
2.2 Change field CommitParentId to CommitId
image

3) JoinEntities doesn`t have Columns for PrimaryKey, composite keys are used instead. It is not bad, but If you have composite primary keys, anyone joining to your table from a child table must "drag along" all those columns, and all those columns are also going to be present in the child table and the JOIN statements are pretty messy. Much better to have a single key for the JOIN
image

4) I do not see any configuration for one to one relationship (maybe I`m missing something):

image

File: backend/Squirrel.Core/Squirrel.Core.DAL/Enums/UserRole.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Squirrel.Core.DAL.Enums;
+
+public enum UserRole
+{
+
```
Comment (alinaSielina) [inline backend/Squirrel.Core/Squirrel.Core.DAL/Enums/UserRole.cs]: You can add 'Admin' for now

File: backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Squirrel.Core.DAL.Enums;
+
+public enum CommentedEntity
+{
+    Project = 1,
```
Comment (alinaSielina) [inline backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs]: Remove 'Project', add 'Script'

**Review by alinaSielina — CHANGES_REQUESTED** [review]
Add configuration to automatically set 'created at' and 'updated at' during the appropriate actions

File: backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs (hunk 2 above)
Comment (ABertashDev) [reply backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs]: CommentedEntity should store names of entities for which we will be able to write comments, like 'Commits', 'Pullrequests' etc.
Maybe you thought about FileTypeEntity which should contain 'Script', 'Readme', weren't you?

Comment (alinaSielina) [reply backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs]: We'll add scripts later, but maybe you're right, we shouldn't add "Script" for now, but "Project" should still be removed

image

File: backend/Squirrel.Core/Squirrel.Core.DAL/Enums/UserRole.cs (hunk 1 above)
Comment (OlegKarapysh) [reply backend/Squirrel.Core/Squirrel.Core.DAL/Enums/UserRole.cs]: done

File: backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs (hunk 2 above)
Comment (OlegKarapysh) [reply backend/Squirrel.Core/Squirrel.Core.DAL/Enums/CommentedEntity.cs]: maybe 'script' is for FileType enum?

**Review by MaxShelestiuk — APPROVED** [review]
All suggested improvements are done

**Review by ABertashDev — APPROVED** [review]

**Review by alinaSielina — APPROVED** [review]

**Pull request merged**